### ISSUE TITLE
bug in modelDef splitVertices method ?

### ISSUE THREAD

**Issue by danielturek** [nimble-dev/nimble#1268]
@perrydv , I think the current incarnation of the `modelDef` class (and associated processing) is largely your work.

This has to do with the `splitVertices` function, which exists in `BUGS_modelDef.R` on line 1637.

I've reduced this to a minimally reproducible example.  I think something is amiss here: model building errors out, when I think it shouldn't.  I began a deep dive into this, but I think it would be quicker and easier for you.

Example below:

```
library(nimble)

dDist <- nimbleFunction(
    run = function(x = double(0),
                   theta1 = double(1),
                   theta2 = double(1),
                   log = integer(0, default = 0)) {
        returnType(double())
        return(0)
    }
)

code <- nimbleCode({
    for (i in 1:2) {
        for (j in 1:2){
            b[i,j] <- X[i, index[j]]
        }
    }
    for (i in 1:2) {
        y[i] ~ dDist(
            theta1 = a[1:2, index[i]],
            theta2 = b[1:2, 1]
        )
    }
})

Rmodel <- nimbleModel(code, constants = list(index=c(1,1)))

## Error in if (is.na(uniqueCurrentVertexIDs[1]) | currentVertexCounts[uniqueCurrentVertexIDs[1]] !=  : 
##  missing value where TRUE/FALSE needed
```

**Comment by paciorek** [comment]
A more minimal example; the user-def dist is not needed.

```
code <- nimbleCode({
    for (i in 1:2) {
        for (j in 1:2){
            b[i,j] <- X[i, index[j]]
        }
    }
    for (i in 1:2) {
        y[i] <- sum(a[1:2, index[i]])
        
    }
})

Rmodel <- nimbleModel(code, constants = list(index=c(1,1)))
```

Also it only seems to happen if `index` is `c(1,1)`. Perhaps a scalar vs. vector issue.

**Comment by paciorek** [comment]
Fixed by #1279
